src/processor/operator/scan/lookup_node_table.cpp

```diff
@@ -0,0 +1,40 @@
+#include "processor/operator/scan/lookup_node_table.h"
+
+using namespace kuzu::common;
+using namespace kuzu::storage;
+
+namespace kuzu {
+namespace processor {
+
+void LookupNodeTable::init(common::nodeID_t nodeID) {
+    nodeIDVector->setValue<nodeID_t>(0, nodeID);
+    hasExecuted = false;
+}
+
+void LookupNodeTable::initLocalStateInternal(ResultSet* resultSet, ExecutionContext* context) {
+    ScanTable::initLocalStateInternal(resultSet, context);
+    for (auto& [_, info] : tableIDToNodeInfo) {
+        info.localScanState = std::make_unique<NodeTableScanState>(info.columnIDs);
+        initVectors(*info.localScanState, *resultSet);
+    }
+}
+
+bool LookupNodeTable::getNextTuplesInternal(ExecutionContext* context) {
+    if (hasExecuted) {
+        return false;
+    }
+    hasExecuted = true;
+    auto transaction = context->clientContext->getTx();
+    auto nodeID = nodeIDVector->getValue<nodeID_t>(0);
+    KU_ASSERT(tableIDToNodeInfo.contains(nodeID.tableID));
+    auto& nodeInfo = tableIDToNodeInfo.at(nodeID.tableID);
+    // TODO(Guodong): The following lines are probably incorrect.
```

Comment:
Can u leave a todo to me to fix the scan of uncommitted part?

Reply:
I might just fix this in my following PR